File: app/src/main/java/br/com/popularmovies/services/movieService/source/MovieDataSource.java
package br.com.popularmovies.services.movieService.source;

import androidx.lifecycle.LiveData;

import java.util.List;

import br.com.popularmovies.data.model.Resource;
import br.com.popularmovies.services.movieService.response.Movie;
import br.com.popularmovies.services.movieService.response.MovieReviews;
import br.com.popularmovies.services.movieService.response.MovieTrailers;
import br.com.popularmovies.services.movieService.response.Movies;

public interface MovieDataSource {

    LiveData<Resource<Movies>> getMovies(String orderBy);

    LiveData<Resource<Movie>> getMovie(int movieId);

    LiveData<Resource<MovieReviews>> getMovieReviews(int movieId);

    LiveData<Resource<Boolean>> saveToFavorites(int movieId, boolean status);

    LiveData<Resource<Void>> saveMovies(List<Movie> movies);

    LiveData<Resource<Void>> saveMovie(Movie movie);

    LiveData<Resource<Void>> removeMovie(Movie movie);

    LiveData<Resource<MovieTrailers>> getMovieTrailers(int movieId);
}

File: app/src/main/java/br/com/popularmovies/services/movieService/source/MovieDataSource.kt
package br.com.popularmovies.services.movieService.source

import br.com.popularmovies.core.network.retrofit.model.Resource
import br.com.popularmovies.services.movieService.response.MovieDto
import br.com.popularmovies.services.movieService.response.MovieReviews
import br.com.popularmovies.services.movieService.response.MovieTrailers
import br.com.popularmovies.services.movieService.response.Movies

interface MovieDataSource {

    suspend fun getMovies(orderBy: String): Resource<Movies>

    suspend fun getMovie(movieId: Int): Resource<MovieDto>

    suspend fun getMovieReviews(movieId: Int): Resource<MovieReviews>

    suspend fun saveToFavorites(movieDto: MovieDto): Resource<Unit>

    suspend fun getMovieTrailers(movieId: Int): Resource<MovieTrailers>
}
